### ISSUE TITLE
addAttribute() helper function overwrites blocktype attributes settings

### ISSUE THREAD

**Issue by elliotcondon** [WordPress/gutenberg#15240]
**Describe the bug**
When registering a block type that supports "anchor", the `attributes.anchor` setting is being overwritten by a helper function [addAttribute()](https://github.com/WordPress/gutenberg/blob/87d7fe4be406aedb3aefa7d961ac3781ab600654/packages/block-editor/src/hooks/anchor.js#L36).

This helper function is designed to work with a static block type that returns an element from the `save()` function but is causing issues with dynamic block types that return `null`.

As an example, here is a dynamic block type that allows "anchor" and expects the anchor value to be serialized within the block HTML JSON.
```
// Register with WP.
var result = wp.blocks.registerBlockType( 'test/my-block', {
	title: 'Test block',
	attributes: {
		anchor: {
			type: 'string'
		}
	}
	supports: {
		anchor: true
	}
	save: function(){
		return null;
	}
});
```

Even though the anchor attribute has been defined for the block type, the result will actually look like this because the `attributes.anchor` setting has been overwritten by the [addAttribute()](https://github.com/WordPress/gutenberg/blob/87d7fe4be406aedb3aefa7d961ac3781ab600654/packages/block-editor/src/hooks/anchor.js#L36) helper function.
```
{
	title: 'Test block',
	attributes: {
		anchor: {
			type: 'string',
			source: 'attribute',
			attribute: 'id',
			selector: '*',
		}
	}
	supports: {
		anchor: true
	}
	save: function(){
		return null;
	}
}
```

**Expected behavior**
If the `attributes.anchor` setting has already been specified when registering a block type, it should not be overwritten by the helper function.

**Additional info**
You can use the following line of code to remove the helper function, which allows dynamic block type to save "anchor" data, but obviously causes issues with core, and is not a valid solution:
```
wp.hooks.removeFilter( 'blocks.registerBlockType', 'core/anchor/attribute' );
```

**Comment by sarahquigley** [comment]
+1 on this. Causing big issues on our projects.

**Comment by okadots** [comment]
+1 definitely to resolve this

**Comment by gziolo** [comment]
It looks like something that needs to be fixed. We can follow what already exists for `supports: align`:

https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/hooks/align.js#L78-L81

**Comment by gziolo** [comment]
It should be fixed with #15959.
